### DISCUSSION TITLE
檢驗波動幅度

### DISCUSSION THREAD

**Discussion by ChiuAlfredo** [ChiuAlfredo/options_day_data#26] [研究]
# 探討目標：

1. 結算日對比之前的價格是否波動更劇烈

# 結算日對比之前的價格是否波動更劇烈

### 分組
選擇權tej202101-202104選擇權日資料
結算日為t_0，往前一天為t_1
* 近結算2日內-t_0跟t_1的資料
* 近結算10日內（不含2日內)-t_2至t_9資料

之後會在增加
* 結算日當天跟結算日前5天的所有波動率是否顯著有顯著差異
* 結算日當天跟結算日前一天的所有波動是否顯著有顯著差異

### 分類
以結算日開盤價最為價性的標準
用價性作為標準
### 買權（call option）
$\frac{K}{S}$​ 範圍 | 價性分類
-- | --
$0.94 \leq \frac{K}{S} ​<0.97 $ | 深度價內 (DITM)
$0.97 \leq \frac{K}{S} ​<0.99 $ | 價內 (ITM)
$0.99\leq \frac{K}{S} <1.01 $ | 價平 (ATM)
$1.01\leq \frac{K}{S} ​<1.03 $ | 價外 (OTM)
$1.03 \leq \frac{K}{S} ​<1.06 $ | 深度價外 (DOTM)

### 賣權（put option）
$\frac{K}{S}$​ 範圍 | 價性分類
-- | --
$1.03 \leq \frac{K}{S} ​<1.06 $ | 深度價內 (DITM)
$1.01\leq \frac{K}{S} ​<1.03 $ | 價內 (ITM)
$0.99\leq \frac{K}{S} <1.01 $ | 價平 (ATM)
$0.97 \leq \frac{K}{S} ​<0.99 $ | 價外 (OTM)
$0.94 \leq \frac{K}{S} ​<0.97$ | 深度價外 (DOTM)

分成10組去做計算

## 波動率計算方式

*  $日報酬r_t  = \ln{\frac{當天開盤_i}{前天收盤_i}}$
$s_t :當天收盤價$
$s_{t-1} :當天開盤價$

* 標準差

$$s = \sqrt{\frac{1}{n-1} \sum_{i=1}^{n} (r_t - \bar{t})^2}$$

n:日報酬數量

* 年化波動 
$$\hat{\sigma} = {s}{\sqrt{252}}$$

$252:年交易天數$

## 使用 F 檢定的步驟：

1. **確定假設**：
   - **虛無假設 \( $H_0$ \)**： $近結算0-1日\hat{\sigma}^2=近結算2-9日 \hat{\sigma}^2$
   - **對立假設 \( $H_1$ \)**： $近結算0-1日\hat{\sigma}^2 \not= 近結算2-9日 \hat{\sigma}^2$

2. ** 樣本** :
   - **樣本1**:近結算0-1日年化波動
   - **樣本2**:近結算2-9日年化波動
 
3. **計算 F 值**：

   F 值計算公式如下：

$$
   F = \frac{\hat{\sigma^2_{\text{近結算0-1日}}}}{\hat{\sigma^2_{\text{近結算2-9日}}}}
$$

4. **拒絕域**：
* $\alpha = 0.01 $
* 雙尾： $p>1 - \frac{\alpha}{2}$ or $p<\frac{\alpha}{2}$ 拒絕虛無假設，近結算0-1日與近結算2-9日有顯著差異

*

**Comment by Will150143** [comment]
1.時間切分是結算日當天和哪一個時段，結算日前一天嗎？還是哪幾天
2.是取哪一個價性做檢驗

**Reply by ChiuAlfredo** [reply]
- [x] 1.時間切分是結算日當天和哪一個時段，結算日前一天嗎？還是哪幾天
- [x] 2.是取哪一個價性做檢驗

**Reply by ChiuAlfredo** [reply]
1.時間切分是結算日當天和哪一個時段，結算日前一天嗎？還是哪幾天
取結算日跟前10天
2.是取哪一個價性做檢驗
所有價性都會做檢驗

**Comment by Will150143** [comment]
1.單一價性下會有不同標的，例如價平會有三檔左右，要如何計算變異數，三檔取平均嗎？
2.每日波動率的公式是TEJ用的公式嗎

**Reply by ChiuAlfredo** [reply]
1. 可能使用算數平均波動率，來算出平均波動
2. 不是，可以看上面 [波動率計算公式]

**Comment by TsuYiii** [comment]
1.比較波動度的方式為以結算日前十天與結算日當天逐一比較，還是是以結算日前十天為一組與結算日當天進行比較呢？
2.在結算日樣本的自由度與非結算日樣本的自由度的公式是否未標示清楚呢？

**Reply by ChiuAlfredo** [reply]
1. 會用前十天為一組與結算日進行比較
2. 已經標記上

**Comment by ChiuAlfredo** [comment]
問題：
1. 波動率公式是否正確，結算日的最低價格很低，所有波動率會很高，例如在結算日時最高價10，最低價0.1這樣就是1000%的波動率

**Comment by ChiuAlfredo** [comment]
<img width="792" alt="image" src="https://github.com/ChiuAlfredo/options_day_data/assets/91011574/f769cc7b-7acd-46ce-b124-6b12abd34120">

幅度：$ (成交價-昨日收盤價)/昨日收盤價 $
振幅：$ (最高價-最低價)/昨日收盤價 $

**Comment by ChiuAlfredo** [comment]
我們要驗證結算日當天的波動跟前十天的波動比較，是否有顯著的差異

實驗中會用價性作為分組標準
### 買權（call option）
$\frac{K}{S}$​ 範圍 | 價性分類
-- | --
$0.94 \leq \frac{K}{S} ​<0.97 $ | 深度價內 (DITM)
$0.97 \leq \frac{K}{S} ​<0.99 $ | 價內 (ITM)
$0.99\leq \frac{K}{S} <1.01 $ | 價平 (ATM)
$1.01\leq \frac{K}{S} ​<1.03 $ | 價外 (OTM)
$1.03 \leq \frac{K}{S} ​<1.06 $ | 深度價外 (DOTM)

### 賣權（put option）
$\frac{K}{S}$​ 範圍 | 價性分類
-- | --
$1.03 \leq \frac{K}{S} ​<1.06 $ | 深度價內 (DITM)
$1.01\leq \frac{K}{S} ​<1.03 $ | 價內 (ITM)
$0.99\leq \frac{K}{S} <1.01 $ | 價平 (ATM)
$0.97 \leq \frac{K}{S} ​<0.99 $ | 價外 (OTM)
$0.94 \leq \frac{K}{S} ​<0.97$ | 深度價外 (DOTM)

分成10組去做計算


## 波動率計算方式


* 算數平均波動率
$$\overline{\sigma} = \frac{1}{n} \sum_{i=1}^{n} \sigma_i$$
其中：

$\overline{\sigma}$ 是算術平均波動率
$\sigma_i$ 是每個標的的波動率
$n$ 是標的數量（在您的例子中，大約是3）

* 變異數
  - 結算日樣本的自由度： 
  $df_1 = n_1 - 1$
  - 非結算日樣本的自由度：
  $df_2 = n_2 - 1$

$$s^2_{\text{結算}} = \frac{1}{df_1} \sum_{i=1}^{n} (\sigma_i - \overline{\sigma})^2$$

$$s^2_{\text{非結算}} = \frac{1}{df_2} \sum_{i=1}^{n} (\sigma_i - \overline{\sigma})^2$$


在裡面我有一個問題是關於我們這樣的比較的每日波動率要如何做計算，目前有找到一些可能的公式：
1. 波動率：(當天高價 - 當天低價) / 當天低價 
2. 幅度：(成交價-昨日收盤價)/昨日收盤價
3. 振福：(最高價-最低價)/昨日收盤價

想請問這三個哪一個公式可以用來比較結算日當天的波動跟前十天的波動比較，是否有顯著的差異

**Comment by ChiuAlfredo** [comment]
報酬率:$ \log{當天收盤/前天收盤}$

**Comment by ChiuAlfredo** [comment]
[檢驗波動幅度 branch](https://github.com/ChiuAlfredo/options_day_data/tree/%E6%AA%A2%E9%A9%97%E6%B3%A2%E5%8B%95%E5%B9%85%E5%BA%A6%2326)

./驗證波動幅度/

* f_test完成
* report.md 輸出f_test結果 [report.md](https://github.com/ChiuAlfredo/options_day_data/blob/05aae4ae7668f5722241214ea39961fc4979d068/%E9%A9%97%E8%AD%89%E6%B3%A2%E5%8B%95/report.md)
* [程式](https://github.com/ChiuAlfredo/options_day_data/blob/05aae4ae7668f5722241214ea39961fc4979d068/%E9%A9%97%E8%AD%89%E6%B3%A2%E5%8B%95/%E9%A9%97%E8%AD%89%E6%B3%A2%E5%8B%95.py)
*

**Comment by Will150143** [comment]
1.這個驗證結果太漂亮了吧
2.我似乎沒有在程式裡看到年化的過程，在說明上會不會有什麼影響
3.自由度的算法是？我有看到 `df_numerator = len(df_0) - 1`，但想說數字有點大，我的理解是在這個價性內用了多少個報酬率-1

**Reply by ChiuAlfredo** [reply]
2. 請看[ /\sqrt(252)](https://github.com/ChiuAlfredo/options_day_data/blob/05aae4ae7668f5722241214ea39961fc4979d068/%E9%A9%97%E8%AD%89%E6%B3%A2%E5%8B%95/%E9%A9%97%E8%AD%89%E6%B3%A2%E5%8B%95.py#L99-L100)
3. 可以直接從
[自由度](https://github.com/ChiuAlfredo/options_day_data/blob/05aae4ae7668f5722241214ea39961fc4979d068/%E9%A9%97%E8%AD%89%E6%B3%A2%E5%8B%95/report.md?plain=1#L8-L11)查看

**Comment by ChiuAlfredo** [comment]
輸出結果
https://mailntustedutw-my.sharepoint.com/:x:/g/personal/b11033003_ms_ntust_edu_tw/ESfJ8fnyUv1Dsinx09x1IaIB3JYMoprJrAcapODLCu20fQ?e=UJnr8o
